### ISSUE TITLE
Failed to read packet: Got error reading packet from server: The slave is connecting using CHANGE MASTER TO MASTER_AUTO_POSITION

### ISSUE THREAD

**Issue by sankarnaanal** [google/mysql-ripple#21]
I1008 09:31:33.836370 73346 mysql_master_session.cc:182] Master session starting
I1008 09:31:34.758591 73346 mysql_client_connection.cc:148] connected to host: x.x.x.x, port: 3306
I1008 09:31:35.422768 73346 mysql_master_session.cc:137] Connected to host: x.x.x.x, port: 3306, server_id: 604687941, server_name:
I1008 09:31:35.691968 73346 mysql_master_session.cc:202] master has semi sync enabled
I1008 09:31:35.692040 73346 mysql_master_session.cc:207] start replicating from ''
I1008 09:31:36.161306 73346 mysql_master_session.cc:230] Master session entering main loop
E1008 09:31:36.382496 73346 mysql_master_session.cc:297] Failed to read packet: Got error reading packet from server: The slave is connecting using CHANGE MASTER TO MASTER_AUTO_POSITION = 1, but the master has purged binary logs containing GTIDs that the slave requires.
I1008 09:31:36.382568 73346 binlog.cc:675] Connection closed last position binlog file: binlog.000000:244, gtid: 0-0-0
I1008 09:31:37.055959 73346 mysql_client_connection.cc:148] connected to host: x.x.x.x, port: 3306
I1008 09:31:37.710474 73346 mysql_master_session.cc:137] Connected to host: x.x.x.x, port: 3306, server_id: 604687941, server_name:
I1008 09:31:37.929569 73346 mysql_master_session.cc:202] master has semi sync enabled
I1008 09:31:37.929639 73346 mysql_master_session.cc:207] start replicating from ''
I1008 09:31:38.364151 73346 mysql_master_session.cc:230] Master session entering main loop
E1008 09:31:38.582463 73346 mysql_master_session.cc:297] Failed to read packet: Got error reading packet from server: The slave is connecting using CHANGE MASTER TO MASTER_AUTO_POSITION = 1, but the master has purged binary logs containing GTIDs that the slave requires.
I1008 09:31:38.582531 73346 binlog.cc:675] Connection closed last position binlog file: binlog.000000:244, gtid: 0-0-0
I1008 09:31:39.254842 73346 mysql_client_connection.cc:148] connected to host: x.x.x.x, port: 3306
I1008 09:31:39.916255 73346 mysql_master_session.cc:137] Connected to host: x.x.x.x, port: 3306, server_id: 604687941, server_name:

**Comment by sankarnaanal** [comment]
I have a master SQL instance in google cloud, while I try to set up a bin log server using MySQL-ripple I'm facing the issue, also I need to set up a slave from bin log server is this possible?

**Comment by pivanof** [comment]
You definitely can set up slave from Ripple. Just execute CHANGE MASTER TO on the replica and use Ripple's IP and port as MASTER_HOST and MASTER_PORT in the statement, also use MASTER_AUTO_POSITION = 1. With that the replica should be able to successfully start replicating from Ripple.

Regarding the issue that Ripple has trying to replicate from the master: it's not clear for me why this could happen. Can you post excerpt from the master's error log? I think it should have more info on what's happening.

**Comment by sankarnaanal** [comment]
but im facing the above issue while running the following command in MySQL-ripple folder /Bazel-bin/rippled -ripple_datadir=./data -ripple_master_address... also I'm using the master instance in google cloud

**Comment by pivanof** [comment]
Right. And the problem is that for some reason the master thinks that Ripple asks to start replicating with a certain GTID that is not in the master's binlogs. I think mysql.err log on the master will have more details about what it doesn't like. Can you post those logs here? Here's documentation on how to find and export the logs: https://cloud.google.com/sql/docs/mysql/instance-info#mysqllogs.

**Comment by sankarnaanal** [comment]
Sure, I will post the logs here.

**Comment by sankarnaanal** [comment]
{
 insertId: "s=35b0979499e449f597024875fb0def4a;i=2651a2;b=3df4b4a8e7c441b7af3023f4be28012c;m=23e2252d274;t=59488d134f425;x=2e839cd52ec7c6ad-0@a1"  
 logName: "projects/project-178317/logs/cloudsql.googleapis.com%2Fmysql.err"  
 receiveTimestamp: "2019-10-10T06:50:31.380054318Z"  
 resource: {…}  
 severity: "ERROR"  
 textPayload: "2019-10-10T06:50:29.924655Z 1105466 [Note] my server_id: 604687941, my uuid: b5105251-e63e-11e8-9d29-42010a800494 slave with server_id: 112211 uuid:  connects with GTID_executed: "  
 timestamp: "2019-10-10T06:50:29.924901Z"  
}

**Comment by pivanof** [comment]
I was able to reproduce this. It looks like we have assumed that MySQL will treat empty gtid_executed as "start replicating from the first binlog file you have", but MySQL doesn't do that. I'll look if it's possible to fix this, but for now please take some value of gtid_executed from the master (either just as the current value of the variable or the GTID list from the first binlog file) and pass it to -ripple_requested_start_gtid_position flag. Then Ripple will be able to successfully start replicating from that position onward.

**Comment by sankarnaanal** [comment]
I will try it
